builtInFunctions/multiESDTNFTTransfer.go

```diff
@@ -0,0 +1,452 @@
+	shardCoordinator vmcommon.Coordinator
+	gasConfig        vmcommon.BaseOperationCost
+	mutExecution     sync.RWMutex
+}
+
+const argumentsPerTransfer = uint64(3)
+
+// NewESDTNFTMultiTransferFunc returns the esdt NFT multi transfer built-in function component
+func NewESDTNFTMultiTransferFunc(
+	funcGasCost uint64,
+	marshalizer vmcommon.Marshalizer,
+	pauseHandler vmcommon.ESDTPauseHandler,
+	accounts vmcommon.AccountsAdapter,
+	shardCoordinator vmcommon.Coordinator,
+	gasConfig vmcommon.BaseOperationCost,
+	activationEpoch uint32,
+	epochNotifier vmcommon.EpochNotifier,
+) (*esdtNFTMultiTransfer, error) {
+	if check.IfNil(marshalizer) {
+		return nil, ErrNilMarshalizer
+	}
+	if check.IfNil(pauseHandler) {
+		return nil, ErrNilPauseHandler
+	}
+	if check.IfNil(accounts) {
+		return nil, ErrNilAccountsAdapter
+	}
+	if check.IfNil(shardCoordinator) {
+		return nil, ErrNilShardCoordinator
+	}
+	if check.IfNil(epochNotifier) {
+		return nil, ErrNilEpochHandler
+	}
+
+	e := &esdtNFTMultiTransfer{
+		keyPrefix:        []byte(vmcommon.ElrondProtectedKeyPrefix + vmcommon.ESDTKeyIdentifier),
+		marshalizer:      marshalizer,
+		pauseHandler:     pauseHandler,
+		funcGasCost:      funcGasCost,
+		accounts:         accounts,
+		shardCoordinator: shardCoordinator,
+		gasConfig:        gasConfig,
+		mutExecution:     sync.RWMutex{},
+		payableHandler:   &disabledPayableHandler{},
+	}
+
+	e.baseEnabled = &baseEnabled{
+		function:        vmcommon.BuiltInFunctionESDTNFTAddURI,
+		activationEpoch: activationEpoch,
+		flagActivated:   atomic.Flag{},
+	}
+
+	epochNotifier.RegisterNotifyHandler(e)
+
+	return e, nil
+}
+
+// SetPayableHandler will set the payable handler to the function
+func (e *esdtNFTMultiTransfer) SetPayableHandler(payableHandler vmcommon.PayableHandler) error {
+	if check.IfNil(payableHandler) {
+		return ErrNilPayableHandler
+	}
+
+	e.payableHandler = payableHandler
+	return nil
+}
+
+// SetNewGasConfig is called whenever gas cost is changed
+func (e *esdtNFTMultiTransfer) SetNewGasConfig(gasCost *vmcommon.GasCost) {
+	if gasCost == nil {
+		return
+	}
+
+	e.mutExecution.Lock()
+	e.funcGasCost = gasCost.BuiltInCost.ESDTNFTTransfer
+	e.gasConfig = gasCost.BaseOperationCost
+	e.mutExecution.Unlock()
+}
+
+// ProcessBuiltinFunction resolves ESDT NFT transfer roles function call
+// Requires the following arguments:
+// arg0 - destination address
+// arg1 - number of tokens to transfer
+// list of (tokenID - nonce - quantity) - in case of ESDT nonce == 0
+// function and list of arguments for SC Call
+// if cross-shard, the rest of arguments will be filled inside the SCR
+// arg0 - number of tokens to transfer
+// list of (tokenID - nonce - quantity/ESDT NFT data)
+// function and list of arguments for SC Call
+func (e *esdtNFTMultiTransfer) ProcessBuiltinFunction(
+	acntSnd, acntDst vmcommon.UserAccountHandler,
+	vmInput *vmcommon.ContractCallInput,
+) (*vmcommon.VMOutput, error) {
+	e.mutExecution.RLock()
+	defer e.mutExecution.RUnlock()
+
```

Comment:
The integration will be different - as deciding whether one call is of BuiltInCall or SmartContract processing or normal transfer is done at the transaction coordinator level and in Arwen -> because of this - the enabled / not enabled part has to be added there. That is why I added the new interface of IsActive

Reply:
ok